runtime/tests/checker/attachments_test.go

```diff
@@ -4665,6 +4665,65 @@ func TestCheckAttachmentPurity(t *testing.T) {
 			}`,
 		)
 
+		require.NoError(t, err)
+	})
+
+	t.Run("remove from global", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			struct S {}
+			attachment Test for S {}
+			var s: S = S()
+			view fun foo() {
+				remove Test from s
```

Comment:
Are the semantics for `remove` on a value-type (e.g. struct) that the variable is re-assigned? Is `s` after the removal a new copy of `s` without the attachment (and the original value is unmodified, but "lost")?

Reply:
The variable is mutated by the `remove`, so a globally scoped or field-scoped struct will be modified.